lib/internal/readline/eventsToAsyncIteratorFactory.js

```diff
@@ -0,0 +1,107 @@
+'use strict';
+const {
+  Promise,
+  SymbolAsyncIterator,
+  ArrayPrototypeConcat,
+} = primordials;
+const FixedQueue = require('internal/fixed_queue');
+
+const PAUSE_THRESHOLD = 1024;
+const RESUME_THRESHOLD = 1;
+const ITEM_EVENTS = ['data'];
+const CLOSE_EVENTS = ['close', 'end'];
+const ERROR_EVENTS = ['error'];
+
+
+function waitNext(emitter, next, events) {
+  return new Promise((resolve, reject) => {
+    const resolveNext = () => {
+      for (let i = 0; i < events.length; i++)
+        emitter.off(events[i], resolveNext);
```

Comment:
You can't remove the `'error'` event otherwise an error that happens while we are not reading a chunk will cause the process to crash.

Reply:
I actually have another listener for the error event on line 72
```js
  function turn(onOff) {
    for (let i = 0; i < closeEvents.length; i++)
      readable[onOff](closeEvents[i], onClose);
    for (let i = 0; i < itemEvents.length; i++)
      readable[onOff](itemEvents[i], onItem);
    for (let i = 0; i < itemEvents.length; i++)
      readable[onOff](errorEvents[i], onError);
  }

  turn('on');
```
Can the process crash removing this listener even having that?
Notice that I remove the error event on the error and the close listener there too.

Reply:
ok thx for the explanation